api/config/esgateway/config_request.go

```diff
@@ -89,6 +91,20 @@ func (c *ConfigRequest) Validate() error {
 // to start the service.
 func (c *ConfigRequest) PrepareSystemConfig(creds *ac.TLSCredentials) (ac.PreparedSystemConfig, error) {
 	c.V1.Sys.Tls = creds
+
+	enableSystemNameServer := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetEnableSystemNameservers().GetValue()
+	nameServers := c.GetV1().GetSys().GetNgx().GetMain().GetResolvers().GetNameservers()
+	if ((nameServers == nil) || (len(nameServers) == 0)) && enableSystemNameServer {
```

Comment:
In Go, the length of a nil slice is 0, so I believe you can simplify this conditional to:

```suggestion
	if len(nameServers) == 0 && enableSystemNameServer {
```

See https://play.golang.org/p/BqOVEFB8YSp